src/main/java/pl/lukakan/showstracker/show/model/Movie.java

```diff
@@ -0,0 +1,117 @@
+package pl.lukakan.showstracker.show.model;
+
+import org.springframework.format.annotation.DateTimeFormat;
+import pl.lukakan.showstracker.cast.role.model.Role;
+
+import javax.persistence.*;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+
+@Entity
+public class Movie {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    @DateTimeFormat(pattern = "yyyy-MM-dd")
+    private LocalDate premiereDate;
+    private String description;
+    private Double averageRate;
+    private Integer numberOfRates;
+    @OneToMany(mappedBy = "movie")
+    private List<Role> roles;
+    @ManyToMany(mappedBy = "movies")
+    private Set<Genre> genres;
+
+    public Movie() {
+
+    }
+
+    public void addGenre(Genre genre) {
```

Comment:
To (używanie this przy dodawaniu) mi wygląda na bardzo niebezpieczną operację, ale chyba nawet tego nie wykorzystujesz - do usunięcia